sklearn/preprocessing/_target_encoder_fast.pyx

```diff
@@ -0,0 +1,165 @@
+cimport cython
+from libc.math cimport isnan
+from libc.string cimport memset
+from libcpp.vector cimport vector
+
+cimport numpy as cnp
+import numpy as np
+
+cnp.import_array()
```

Comment:
Do we really need this here? It seems that we only use typed memory views in this code. Or maybe this is necessary to call `np.empty`?

Reply:
`cnp` is still used for the dtypes and I think it's safer to include whenever we cimport `cnp`. I commented about this recently in https://github.com/scikit-learn/scikit-learn/pull/25780#issuecomment-1460865458 and da-woods from Cython recommends it: https://groups.google.com/g/cython-users/c/6qGmzvIv7Io

Reply:
Indeed:

> My summary recommendation is: if you cimport Numpy the safest thing is to always call import_array(). The cases where it isn't needed depend on Cython version (and possibly Numpy version) so that's why I'm recommending that it's always needed. I do plan to go through the Cython docs and update it.

Maybe @jjerphan could be expanded on this topic :)

Reply:
Based on remark of Cython developers, we agreed that calling `cnp.import_array` is safer.

---

FYI, this is something we recently have discussed in https://github.com/scikit-learn/scikit-learn/pull/25780.

Side-note: some calls to `cnp.import_array` might have been removed before this discussion.